source/OutgoingMessages.Application/UseCases/DelegateMessage.cs

```diff
@@ -83,6 +83,16 @@ public async Task<OutgoingMessage> DelegateAsync(
             cancellationToken)
             .ConfigureAwait(false);
 
+        if (messageDelegation != null)
+        {
+            _logger.LogInformation(
+                "Message was delegated from {FromActorNumber} to {ToActorNumber}, with role {ActorRole} based on delegation with sequence number {SequenceNumber}",
+                delegatedByActorNumber.Value,
+                messageDelegation.DelegatedTo.ActorNumber.Value,
+                delegatedByActorRole.Code,
+                messageDelegation.SequenceNumber);
+        }
+
         return messageDelegation is not null
             ? Receiver.Create(messageDelegation.DelegatedTo.ActorNumber, messageDelegation.DelegatedTo.ActorRole)
             : null;
```

Comment:
Could we log the receiver instead? 
- Receiver.Create() could potentiel do should else.

```suggestion
var receiver = messageDelegation is not null
            ? Receiver.Create(messageDelegation.DelegatedTo.ActorNumber, messageDelegation.DelegatedTo.ActorRole)
            : null
        if (receiver != null)
        {
            _logger.LogInformation(
                "Message was delegated from {FromActorNumber} to {ToActorNumber}, with role {ActorRole} based on delegation with sequence number {SequenceNumber}",
                delegatedByActorNumber.Value,
                receiver.ActorNumber.Value,
                 receiver.ActorRode.Name,
                messageDelegation.SequenceNumber);
        }

        return receiver;
```

Reply:
please see my last commit 👍